app/process/handler/consensus-message/consensus-message.go

```diff
@@ -120,6 +133,65 @@ func (cmh ConsensusMessageHandler) handleEthTxMessage(m *validatorproto.TopicEth
 	return cmh.scheduler.Cancel(m.TransactionId)
 }
 
+func (cmh ConsensusMessageHandler) verifyEthTxAuthenticity(m *validatorproto.TopicEthTransactionMessage) (bool, error) {
+	tx, _, err := cmh.ethereumClient.Client.TransactionByHash(context.Background(), common.HexToHash(m.EthTxHash))
+	if err != nil {
+		cmh.logger.Warnf("[%s] - Failed to get eth transaction by hash [%s]. Error [%s]", m.TransactionId, m.EthTxHash, err)
+		return false, err
+	}
+
+	if strings.ToLower(tx.To().String()) != strings.ToLower(cmh.bridgeContractAddress) {
+		return false, nil
+	}
+
+	txMessage, signatures, err := ethhelper.DecodeBridgeMintFunction(tx.Data())
+	if err != nil {
+		return false, err
+	}
+
+	if txMessage.TransactionId != m.TransactionId {
+		return false, nil
+	}
+
+	dbTx, err := cmh.transactionRepository.GetByTransactionId(m.TransactionId)
+	if err != nil {
+		return false, err
+	}
+	if dbTx == nil {
+		return false, nil
+	}
```

Comment:
Can we merge these if statements?
In any way we are returning false as first return paratemer and whatever the error value is. (Might be null, might be a specific error)

Reply:
Well, not really. The first error returns the actual error, the second returns `false, nil`, which is not an actual error, just that it's actually `invalid`.